ztp/siteconfig-generator/siteConfig/siteConfig.go

```diff
@@ -228,6 +231,57 @@ type TangConfig struct {
 	Thumbprint string `yaml:"thumbprint" json:"thp"`
 }
 
+type DiskPartition struct {
+	Device     string       `yaml:"device"`
+	Partitions []Partitions `yaml:"partitions"`
+}
+
+type Partitions struct {
+	MountPoint    string `yaml:"mount_point" `
+	Size          int    `yaml:"size"`
+	Start         int    `yaml:"start"`
+	MountFileName string `yaml:"-"`
+	Label         string `yaml:"-"`
+}
+
+type PartitionsUserInput Partitions
```

Comment:
Since this alias is really designed to be used in `UnmarshalYAML`, we should scope it in that function so it doesn't get used outside of its intended purpose.

Reply:
Moved it into the method and updated the tests cases